src/globals.cpp

```diff
@@ -450,6 +475,10 @@ CThinBlockData thindata;
 CGrapheneBlockData graphenedata;
 CCompactBlockData compactdata;
 ThinTypeRelay thinrelay;
+std::map<CNode *, CMempoolSyncState> mempoolSyncRequested;
+std::map<CNode *, CMempoolSyncState> mempoolSyncResponded;
+std::chrono::time_point<std::chrono::high_resolution_clock> lastMempoolSync = std::chrono::high_resolution_clock::now();
+CCriticalSection cs_mempoolsync;
```

Comment:
the map definition should include GUARDED_BY(cs_mempoolsync);  see other examples in globals.  Also you'll have then move CCriticalSection cs_mempoolsync to be before any guarded by statements.